### PR TITLE
CORE-549, CORE-634 Flow resources

### PR DESCRIPTION
**Pull request by object-required**

### CODE REVIEW

File: cypress/e2e/flow-builder/undo-redo/block-resources.cy.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import {UNDO_REDO_WAIT_MS} from '../../../support/commands'
+
+describe('Block resources', () => {
+  beforeEach(() => {
+    cy.createFlow({
+      label: 'Undo-redo test',
```
Comment (safydy-r) [inline cypress/e2e/flow-builder/undo-redo/block-resources.cy.js:6]: maybe "undo-redo block resources"

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -118,6 +119,26 @@ Cypress.Commands.add('selectBlock', (uuid: string) => {
   })
 })
 
+Cypress.Commands.add('setBlockText', (language: string, channel: string, value: string) => {
```
Comment (safydy-r) [inline cypress/support/commands.ts:122]: I'd suggest we rename it to "setBlockTextResource"

**Review by safydy-r — CHANGES_REQUESTED** [review]
in general it's good, left minor change requests. Let me know if you have a question

**Review by safydy-r — APPROVED** [review]
LGTM

File: cypress/support/commands.ts [hunk 3]
```diff
@@ -137,6 +138,26 @@ Cypress.Commands.add('selectBlock', (uuid: string) => {
...
+
+  cy.get('@resourceEditor')
+    .find(`[data-cy="resource--${channel}"]`)
+    .as('editor')
+    .type(value)
+
+  cy.wait(UNDO_REDO_WAIT_MS)
+
+  return cy.get('@editor').then((editor) => {
+    return cy.wrap(editor)
```
Comment (bulatgab) [inline cypress/support/commands.ts:157]: 👍

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 4]
```diff
@@ -11,6 +11,7 @@
           :rows="rows"
           :valid-state="isValid"
           :disabled-auto-complete="disabledAutoComplete"
+          :cypress-data-attribute="`resource--${mode}`.toLowerCase()"
```
Comment (bulatgab) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:14]: When do we use `cypress-data-attribute` vs `data-cy`?

File: cypress/e2e/flow-builder/undo-redo/block-resources.cy.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        .find('textarea')
+        .should('have.value', '')
+
+      cy.redo()
+      cy.get('@choicesBuilder')
+        .find('textarea')
+        .should('have.value', sampleChoice)
+    })
+  })
+})
```
Comment (bulatgab) [inline cypress/e2e/flow-builder/undo-redo/block-resources.cy.js:51]: 👍

**Pull request merged**